### DISCUSSION TITLE
Ignore / Suppress specific magic number warning in detekt baseline.xml for a Compose function

### DISCUSSION THREAD

**Discussion by talhahasanzia** [detekt/detekt#7274] [Q&A] [answered]
To ignore magic number in a class, we can do this:
```
    <ID>MagicNumber:FeatureManager.kt$FeatureManager$30</ID>
```

Where, `FeatureManager` is class, and `30` is magic number.

What would happen if this is in compose lets say:

```
@Composable
fun CustomComposable(){
    LaunchedEffect(Unit) {
        delay(3000) // this needs to be added to baseline
        ...
    }
}
```

I have tried multiple approaches, but still I am unable to suppress this.

Referencing direct function from file reference:

```
    <ID>MagicNumber:CustomComposable.kt$CustomComposable$3000</ID>
```

Referencing direct magic number from file reference:
```
    <ID>MagicNumber:CustomComposable.kt$3000</ID>
```

Referencing direct magic number from element (function) reference:
```
    <ID>MagicNumber:CustomComposable$3000</ID>
```

This is hit and trial. I know that already, that this is not the way to do it (not mentioned in the docs what to do either)

Only reference to read for Baseline I found is this [detekt website](https://detekt.dev/docs/introduction/baseline/)

**Comment by BraisGabin** [comment, marked as answer]
In general I don't recommend you to create the baseline by hand. I would recommend to execute the baseline task to generate it for you. If there are more lines that you want you can simply remove those.

If you still want to build it by hand I recommend you to go to the `txt` report and copy the lines from there. They follow the same pattern, I don't even know exactly which one it is.

**Reply by talhahasanzia** [reply]
Thanks, 
To sum it up, instead of:
```
    <ID>MagicNumber:CustomComposable.kt$CustomComposable$3000</ID>
```
We should do:
```
    <ID>MagicNumber:CustomComposable.kt$3000</ID>
```
Just refer magic number in case of Composable with its file.
We can find the exact signature here: `build/reports/detekt/detekt.txt`

Example report entry:
```
MagicNumber - [<anonymous>] at /Users/Documents/androidApp/src/main/java/tech/android/ui/BottomSheet.kt:170:33 - Signature=BottomSheet.kt$0.7f
```
Suppress format (as per [detekt](https://detekt.dev/docs/introduction/baseline)) : `<RuleID>:<Codesmell_Signature>`

Finally:
```
<ID>MagicNumber:BottomSheet.kt$0.7f<ID>
```
